components/script/dom/htmltablecellelement.rs

```diff
@@ -123,6 +128,20 @@ impl HTMLTableCellElementLayoutHelpers for LayoutJS<HTMLTableCellElement> {
                 .unwrap_or(LengthOrPercentageOrAuto::Auto)
         }
     }
+
+    fn get_inherited_cellpadding(&self) -> Option<u32>
+    {
+        unsafe {
+            // Follow chain: Cell -> Row -> Section -> Table
+            self.upcast::<Node>().parent_node_ref()
```

Comment:
Note: https://html.spec.whatwg.org/#tables-2:attr-table-cellpadding is the spec we're using, not the w3c's specification.